js/main.js

```diff
@@ -40,5 +47,82 @@ function submitInfo(event) {
   $entryForm.reset();
 }
 
+// function to render entries
+function renderEntry(entry) {
+  // creates elements of DOM tree:
+  const $row = document.createElement('div');
+  $row.className = 'row entry-row';
+  const $column = document.createElement('div');
+  $column.className = 'column-full entry-column';
+  const $entryContainer = document.createElement('div');
+  $entryContainer.className = 'entry-container';
+  const $list = document.createElement('ul');
+  $list.className = 'journal-entry';
+  const $listItem = document.createElement('li');
+  $listItem.className = 'user-entry';
+  const $img = document.createElement('img');
+  $img.setAttribute('src', entry.url);
+  $img.className = 'entry-img';
+  const $textContainer = document.createElement('div');
+  $textContainer.className = 'text-container';
+  const $title = document.createElement('h2');
+  $title.className = 'entry-title';
+  $title.textContent = entry.title;
+  const $notes = document.createElement('p');
+  $notes.className = 'entry-notes';
+  $notes.textContent = entry.notes;
+
+  // append DOM nodes to build tree
+
+  $row.appendChild($column);
+  $column.appendChild($entryContainer);
+  $entryContainer.appendChild($list);
+  $list.appendChild($listItem);
+  $listItem.appendChild($textContainer);
+  $listItem.appendChild($img);
+  $textContainer.appendChild($title);
+  $textContainer.appendChild($notes);
+
+  // prepends DOM tree to list
+  $entriesList.appendChild($row);
```

Comment:
Appending to the DOM should happen outside this function. You will append to your `ul`

Reply:
Do I need to put it in the submit callback vs the renderEntries function?

Reply:
It will happen in a couple places. Inside your `submit` callback, and inside your `DOMContentLoaded` callback function